### DISCUSSION TITLE
Error on trying to reinstall

### DISCUSSION THREAD

**Discussion by Dentonknifeworks** [aaronwmorris/indi-allsky#672] [Q&A] [answered]
After the first install, I found for whatever reason my username/password would not work on the config.
Well, I thought I will just run the setup again.
I now get

**** Backup DB prior to migration ****
Error: While importing 'app', an ImportError was raised:

Traceback (most recent call last):
  File "/home/pi/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/flask/cli.py", line 218, in locate_app
    __import__(module_name)
  File "/home/pi/indi-allsky/app.py", line 3, in <module>
    from indi_allsky.flask import create_app
  File "/home/pi/indi-allsky/indi_allsky/__init__.py", line 3, in <module>
    from .allsky import IndiAllSky
  File "/home/pi/indi-allsky/indi_allsky/allsky.py", line 30, in <module>
    from .config import IndiAllSkyConfig
  File "/home/pi/indi-allsky/indi_allsky/config.py", line 15, in <module>
    from .flask.models import IndiAllSkyDbConfigTable
  File "/home/pi/indi-allsky/indi_allsky/flask/__init__.py", line 16, in <module>
    from .views import bp_allsky  # noqa: E402
  File "/home/pi/indi-allsky/indi_allsky/flask/views.py", line 26, in <module>
    import cv2
  File "/home/pi/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/cv2/__init__.py", line 181, in <module>
    bootstrap()
  File "/home/pi/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/cv2/__init__.py", line 153, in bootstrap
    native_module = importlib.import_module("cv2")
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXELF: cannot open shared object file: File name too long

Usage: flask [OPTIONS] COMMAND [ARGS]...
Try 'flask --help' for help.

Error: No such command 'db'.

The script exited abnormally, please try to run again..

**Comment by aaronwmorris** [comment]
What distribution and CPU platform are you installing on?

**Reply by Dentonknifeworks** [reply]
Pi with 64 bit OS bullseye full desktop
 pi@indi:~/indi-allsky $ lscpu
Architecture:                    aarch64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
CPU(s):                          4
On-line CPU(s) list:             0-3
Thread(s) per core:              1
Core(s) per socket:              4
Socket(s):                       1
Vendor ID:                       ARM
Model:                           3
Model name:                      Cortex-A72
Stepping:                        r0p3
CPU max MHz:                     1800.0000
CPU min MHz:                     600.0000
BogoMIPS:                        108.00
L1d cache:                       128 KiB
L1i cache:                       192 KiB
L2 cache:                        1 MiB
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Vulnerable
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fp asimd evtstrm crc32 cpuid

**Reply by aaronwmorris** [reply]
Try upgrading you python modules...

```
source virtualenv/indi-allsky/bin/activate

pip3 install --upgrade -r requirements_debian11.txt
```

and rerun setup.sh

**Reply by Dentonknifeworks** [reply]
Same error

**Reply by aaronwmorris** [reply, marked as answer]
Hmm... okay, let's try recreating the virtualenv...

```
deactivate

mv virtualenv/indi-allsky virtualenv/indi-allsky_old
```

Re-run setup.sh.  It should recreate the virtualenv.

**Reply by Dentonknifeworks** [reply]
Ok that got me back up and running what is the best way to reset the Indi-allsky login as it not taking what I intended as a password

**Reply by Dentonknifeworks** [reply]
Well I just found that I needed to Cap the A in admin

**Reply by Dentonknifeworks** [reply]
And just to ask any chance of adding a IMX519? that is what I am running

**Reply by aaronwmorris** [reply]
Sure, I can add the IMX519.  In reality, you can select any of the libcamera cameras and use it.  The debayer algorithm might be wrong, but you can at least verify it works.

**Reply by Dentonknifeworks** [reply]
Yes I am finding that Thank you

**Reply by aaronwmorris** [reply]
Merged #673 with the imx519 metadata.  If you could verify the bayer pattern, I will correct it in my code if it is not BGGR.

**Reply by aaronwmorris** [reply]
Sorry, Merged #674 to complete the integration.  I forgot the web config options.

**Reply by Dentonknifeworks** [reply]
LOL thats what happened I just came back to say something.

**Reply by Dentonknifeworks** [reply]
Listing of the camera
pi@indi:~ $ libcamera-still --list
Available cameras
-----------------
0 : imx519 [4656x3496] (/base/soc/i2c0mux/i2c@1/imx519@1a)
    Modes: 'SRGGB10_CSI2P' : 1280x720 [80.01 fps - (1048, 1042)/2560x1440 crop]
                             1920x1080 [60.05 fps - (408, 674)/3840x2160 crop]
                             2328x1748 [30.00 fps - (0, 0)/4656x3496 crop]
                             3840x2160 [18.00 fps - (408, 672)/3840x2160 crop]
                             4656x3496 [9.00 fps - (0, 0)/4656x3496 crop]

**Reply by aaronwmorris** [reply]
Thanks!  Merged #675 to correct the bayer pattern to RGGB.
